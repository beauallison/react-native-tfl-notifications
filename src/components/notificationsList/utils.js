import React from 'react';
import propTypes from 'prop-types';
import NotificationsItem from './components/notificationsItem';

export const keyExtractor = item => item.ID;

export const renderItem = ({ item }) => ((<NotificationsItem {...item} />));

renderItem.propTypes = {
  item: propTypes.shape({}).isRequired,
};
