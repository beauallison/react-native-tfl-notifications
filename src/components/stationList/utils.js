import React from 'react';
import propTypes from 'prop-types';
import StationItem from './components/stationItem';

export const keyExtractor = item => item.ID;

export const renderItem = ({ item }) => ((<StationItem {...item} />));

renderItem.propTypes = {
  item: propTypes.shape({}).isRequired,
};
