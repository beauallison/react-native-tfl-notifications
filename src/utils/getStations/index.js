const _ = require('lodash');
const getStatus = require('../getStatus');

module.exports = async stations =>
  Promise.all(_.keys(stations).map(async (ID) => {
    const station = stations[ID];
    const status = await getStatus(station)
      .then(res => res.status)
      .catch(() => 'Unknown');

    return {
      ID,
      station,
      status,
    };
  }));
