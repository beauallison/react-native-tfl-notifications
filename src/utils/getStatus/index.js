const _ = require('lodash');
const { API } = require('../../../config');

// GET /Line/{ids}/Status
// https://api.tfl.gov.uk/Line/{id}/Status
module.exports = async (line) => {
  const response = await fetch(`${API}/Line/${line}/Status?detail=true`);
  const body = await response.json();

  return {
    status: _.get(body, '[0].lineStatuses[0].statusSeverityDescription', 'Unknown'),
    severity: _.get(body, '[0].lineStatuses[0].statusSeverity', 'Unknown'),
  };
};

