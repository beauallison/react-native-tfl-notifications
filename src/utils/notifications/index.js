import update from './update';
import store from './store';

module.exports = {
  update,
  store,
};
