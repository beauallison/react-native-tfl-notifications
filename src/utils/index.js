import getStations from './getStations';

module.exports = {
  getStations,
};

