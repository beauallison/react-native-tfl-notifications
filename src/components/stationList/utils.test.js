import { keyExtractor, renderItem } from './utils';

describe('components/stationList/utils', () => {
  describe('keyExtractor', () => {
    it('should extract a key', () =>
      expect(keyExtractor({ ID: 'TEST' })).toEqual('TEST'));
  });

  describe('renderItem', () => {
    it('should render an item', () => {
      const render = renderItem({ item: { station: 'bakerloo', status: 'Good Service' } });

      expect(render).toMatchSnapshot();
    });
  });
});
