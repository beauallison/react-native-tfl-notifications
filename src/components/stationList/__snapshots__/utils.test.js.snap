// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/stationList/utils renderItem should render an item 1`] = `
<StationItem
  station="bakerloo"
  status="Good Service"
/>
`;
